src/test/java/marquez/dao/JobDAOTest.java

```diff
@@ -89,6 +97,39 @@ public void testInsert() {
     assertJobFieldsMatch(job, jobFound);
   }
 
+  @Test(expected = UnableToExecuteStatementException.class)
+  public void testInsert_Duplicate() {
+    UUID newNamespaceId = UUID.randomUUID();
+    insertNamespace(newNamespaceId, "newNsForDupTest", "Amaranta");
+    jobDAO.insert(job);
+
+    try {
+      Job jobWithDiffNsSameName =
+          new Job(
+              UUID.randomUUID(),
+              job.getName(),
+              "location",
+              newNamespaceId,
+              "desc",
+              Collections.<String>emptyList(),
+              Collections.<String>emptyList());
+      jobDAO.insert(jobWithDiffNsSameName);
+    } catch (UnableToExecuteStatementException e) {
+      fail("failed to insert a job with the same name into a different namespace");
+    }
+
+    Job jobWithSameNsSameName =
+        new Job(
+            UUID.randomUUID(),
+            job.getName(),
+            "location",
+            job.getNamespaceGuid(),
+            "desc",
+            Collections.<String>emptyList(),
+            Collections.<String>emptyList());
+    jobDAO.insert(jobWithSameNsSameName);
+  }
+
```

Comment:
This should be two test cases, since you're trying to prove at least two things here.

1. Same job name, diff NS is ok.
2. Same job name, same ns is NOT ok.

The test is trying to do too much and is considerably less clear than two single-intent testes.

Reply:
sure, can break it up